rp2040-hal/src/timer.rs

```diff
@@ -35,21 +34,20 @@ fn release_alarm(mask: u8) {
     });
 }
 
-fn get_counter(timer: &crate::pac::timer::RegisterBlock) -> Instant {
-    let mut hi0 = timer.timerawh.read().bits();
-    let timestamp = loop {
-        let low = timer.timerawl.read().bits();
-        let hi1 = timer.timerawh.read().bits();
-        if hi0 == hi1 {
-            break (u64::from(hi0) << 32) | u64::from(low);
-        }
-        hi0 = hi1;
-    };
-    TimerInstantU64::from_ticks(timestamp)
-}
 /// Timer peripheral
+///
+/// This struct logically wraps a `pac::TIMER`, but doesn't actually store it:
+/// As after initialization all accesses are read-only anyways, the `pac::TIMER` can
+/// be summoned unsafely instead. This allows timer to be cloned.
+///
+/// (Alarms do use write operations, but they are local to the respective alarm, and
+/// those are still owned singletons.)
+///
+/// As the timer peripheral needs to be started first, this struct can only be
+/// constructed by calling `Timer::new(...)`.
```

Comment:
While I see the value of this comment, I am not sure we should let implementation details transpire in the html doc.

Reply:
Oh yes, sorry - I didn't even notice that this is a doc comment.